core/kazoo_media/src/kz_media_recording.erl

```diff
@@ -517,14 +517,24 @@ store_url(#state{doc_db=Db
     Options = [{'plan_override', Handler}],
     kz_media_url:store(Db, {<<"call_recording">>, MediaId}, MediaName, Options).
 
--spec should_store_recording() -> store_url().
--spec should_store_recording(api_binary()) -> store_url().
-should_store_recording(Url) ->
+-spec should_store_recording(ne_binary(), api_binary()) -> store_url().
+should_store_recording(AccountId, Url) ->
     case kz_util:is_empty(Url) of
-        'true' -> should_store_recording();
+        'true' -> maybe_storage_plan(AccountId);
         'false' -> {'true', 'other', Url}
     end.
 
+-spec maybe_storage_plan(ne_binary()) -> store_url().
+maybe_storage_plan(AccountId) ->
+    AccountDb = kz_util:format_account_mod_id(AccountId),
+    Plan = kzs_plan:get_dataplan(AccountDb, <<"call_recording">>),
+    case maps:get('tag', Plan, 'local') =/= 'local'
```

Comment:
no, it has different meanings. {'true', 'local'} means "yes, go ahead and save". `tag := local` is default connection.
if connection is default and no `att_handler` we should look in system_config as before.